src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs

```diff
@@ -275,6 +276,37 @@ public void TestPreviewDiagnostic()
             }
         }
 
+        [Fact, Trait(Traits.Editor, Traits.Editors.Preview)]
+        public void TestPreviewWorkspaceDoesNotLeakSolution()
+        {
+            // Verify that analyzer execution doesn't leak solution instances from the preview workspace.
+
+            var previewWorkspace = new PreviewWorkspace();            
+            Assert.NotNull(previewWorkspace.CurrentSolution);
+            var project = previewWorkspace.CurrentSolution.AddProject("project", "project.dll", LanguageNames.CSharp);
+            Assert.True(previewWorkspace.TryApplyChanges(project.Solution));
+            var solutionObjectReference = ObjectReference.Create(previewWorkspace.CurrentSolution);
+
+            var analyzers = ImmutableArray.Create<DiagnosticAnalyzer>(new CommonDiagnosticAnalyzers.NotConfigurableDiagnosticAnalyzer());
+            ExecuteAnalyzers(previewWorkspace, analyzers);
+
+            previewWorkspace.Dispose();
+            solutionObjectReference.AssertReleased();
```

Comment:
I am explicitly disposing the workspace in the previous line. I am not sure what your recommendation is over here?